src/Raven.Server/Documents/Sharding/ShardedContext.cs

```diff
@@ -17,33 +26,60 @@ public unsafe class ShardedContext
     {
         public const int NumberOfShards = 1024 * 1024;
 
+        public QueryMetadataCache QueryMetadataCache = new QueryMetadataCache();
+
+        private readonly ServerStore _serverStore;
         private DatabaseRecord _record;
         public RequestExecutor[] RequestExecutors;
         private readonly long _lastClientConfigurationIndex;
 
+        private readonly ConcurrentDictionary<string, IndexDefinition> _cachedMapReduceIndexDefinitions = new(StringComparer.OrdinalIgnoreCase);
+        private readonly ConcurrentDictionary<string, AutoIndexDefinition> _cachedAutoMapReduceIndexDefinitions = new(StringComparer.OrdinalIgnoreCase);
+        public readonly ScriptRunnerCache ScriptRunnerCache;
+        private readonly RavenConfiguration _configuration;
+
         private ShardExecutor _shardExecutor;
         public ShardExecutor ShardExecutor => _shardExecutor;
 
         public int[] FullRange;
 
-        public ShardedContext(ServerStore server, RawDatabaseRecord record)
+        public ShardedContext(ServerStore serverStore, DatabaseRecord record)
         {
             DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal, "reduce the record to the needed fields");
             DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal, "Need to refresh all this in case we will add/remove new shard");
 
+            _serverStore = serverStore;
             _record = record;
-            _lastClientConfigurationIndex = server.LastClientConfigurationIndex;
+
+            UpdateMapReduceIndexes(record.Indexes
+                .Where(x => x.Value.Type.IsStaticMapReduce())
+                .ToDictionary(x => x.Key, x => x.Value));
+
+            UpdateAutoMapReduceIndexes(record.AutoIndexes
+                .Where(x => x.Value.Type.IsAutoMapReduce())
+                .ToDictionary(x => x.Key, x => x.Value));
+
+            _configuration = RavenConfiguration.CreateForDatabase(_serverStore.Configuration, DatabaseName);
+
+            foreach ((string key, string value) in _record.Settings)
+                _configuration.SetSetting(key, value);
+
+            _configuration.Initialize();
```

Comment:
This does not handle database record changes at all

Reply:
done